.github/workflows/maven-ci.yml

```diff
@@ -0,0 +1,81 @@
+name: OTP CI Build
+
+on:
+  push:
+  pull_request:
+    branches:
+      - ibi-dev
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    env:
+      MAVEN_OPTS: -Xmx2g
+    steps:
+      # Starting in v2.2 checkout action fetches all tags when fetch-depth=0, for auto-versioning.
+      - uses: actions/checkout@v2.3.2
+        with:
+          fetch-depth: 0
+      # Java setup step completes very fast, no need to run in a preconfigured docker container.
+      - name: Set up JDK 8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      # Retain the local Maven repository to speed up builds.
+      - name: Setup Maven Cache
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: ~/.m2
+          key: maven-local-repo
+      # So that the branch name can be reference via $GITHUB_HEAD_REF_SLUG.
+      - name: Inject slug/short variables
+        uses: rlespinasse/github-slug-action@v3.x
+      # Build and test deployment.
+      - name: Build and Test
+        run: mvn clean verify --batch-mode --show-version --update-snapshots -Dgpg.skip
+      # Only prepare deploy files on push events to avoid duplicate pushes to s3 when a commit is
+      # pushed to a branch with an open pull request.
+      - name: Prepare build files for deployment to s3
+        if: github.event_name == 'push'
+        run: ./scripts/prep-build-for-deploy-from-github-actions.sh
+      # Copy build files to AWS.
+      - name: Copy build files to the otp-repo bucket
+        if: github.event_name == 'push'
+        uses: jakejarvis/s3-sync-action@master
+        with:
+          args: --acl public-read
+        env:
+          AWS_S3_BUCKET: otp-repo
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          SOURCE_DIR: 'deploy'
```

Comment:
I realize that this is nice and easy with just a bit of config, but this is a security risk to send AWS credentials to a third party GitHub action. I would only do this with something built specifically from either GitHub or Amazon. The underlying script that this plugin uses is quite simple, so I don't think it's worth the risk of including this. It'd probably be best to simply create a script in this repo that handles this so there is no risk of credentials being misused by a third party.

Reply:
Reverted back to using a script.